deepchecks/core/check_result.py

```diff
@@ -632,13 +624,19 @@ def to_json(self, with_display: bool = True):
         ----------
         with_display : bool
             controls if to serialize display or not
+            (the parameter is deprecated and does not have any effect since version 0.6.4,
+            it will be removed in future versions)
 
         Returns
         -------
         str
         """
-        # TODO: not sure if the `with_display` parameter is needed
-        # add deprecation warning if it is not needed
+        if with_display is not None:
+            warnings.warn(
+                '"with_display" parameter is deprecated and does not have any effect '
```

Comment:
it will not be removed, it is just that check failure doesnt have a display.
the function needs to be the same or else the suite.to_json would fail

Reply:
didn't you just re-added it?

Reply:
oh not for check fail